dispatcher/backend/src/db/models.py

```diff
@@ -243,10 +304,30 @@ class Task(Base):
     worker: Mapped["Worker"] = relationship(back_populates="tasks", init=False)
 
     @classmethod
-    def get_or_none_by_id(cls, session: Session, id: UUID) -> Optional["Task"]:
-        """Search DB for a task by UUID, returns None if not found"""
+    def check(
+        cls,
+        task: "Task",
+        exception_class: Type[Exception] = Exception,
+        *exception_args: object,
+    ) -> None:
+        """Raise the exception passed in parameters if task is None."""
+        raise_if_none(task, exception_class, *exception_args)
+
+    @classmethod
+    def get(
+        cls,
+        session: Session,
+        id: UUID,
+        exception_class: Type[Exception] = Exception,
+        *exception_args: object,
+    ) -> "Task":
+        """Search DB for a task by UUID
+
+        If the check of the task is not ok, raise thes exception passed in parameters"""
         stmt = select(Task).where(Task.id == id)
-        return session.execute(stmt).scalar_one_or_none()
+        task = session.execute(stmt).scalar_one_or_none()
+        cls.check(task, exception_class, *exception_args)
```

Comment:
If this is ran unconditionally, I see no reason to have a separate method. It seems that we don't call this `.check()` anywhere else so I think we can simplify here